dts/bindings/base/pm.yaml

```diff
@@ -30,3 +30,9 @@ properties:
     description: |
       Automatically configure the device for runtime power management after the
       init function runs.
+
+  reinit-power-states:
+    type: phandles
+    description: |
+      System power states in which a peripheral's configuration is lost, and
+      must be reinitialized to ensure correct operation.
```

Comment:
I feel this should be limited to suspend-to-idle states.
For s2ram states, this should be considered as default.

Reply:
That'd require additional logic to handle and thus add complexity. Since the property describes hardware, I think it's a better idea to have it defined, even for states that may always require it, such as standby.

Reply:
Indeed, I understand the involved complexity, but I guess in most cases, there will be a state where all devices lose context by default, forcing to document it doesn't feel right.
One idea could be to make "standby" default, but this could be error prone for most users. Maybe we can add a cpu property for that if required if "standby" is not standard enough.
@ceolin, @gmarull thoughts?

Reply:
Adding on to this:
As per our discord discussion I agree it makes sense to have this property apply to non-s2ram (and deeper) states only.

This is following your suggestion of aligning with [the Linux definition](https://www.kernel.org/doc/html/next/admin-guide/pm/sleep-states.html#suspend-to-ram) of everything below (and including) the S2RAM inherently requiring reinitialization of peripherals:
> *In many cases, all peripheral buses lose power when entering S2RAM, so devices must be able to handle the transition back to the "on" state.*

I'll be updating the device tree definition description to align with this.